src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,248 @@
+  std::vector<std::vector<std::string>> dataset;
+  boost::string_view token, copy;
+  stringstream streamLine;
+  while (std::getline(fin, line))
+  {
+    streamLine.str(line);
+    dataset.push_back(std::vector<std::string>());
+    // delimeter[0] becase the standard function accepts char as input.
+    while (std::getline(streamLine, word, column_delimiter[0]))
+    {
+      dataset[dataset.size()-1].push_back(word);
+    }
+    streamLine.clear();
+  }
+  size_t col = 0;
+  // Preparing the input dataset on which string manipulation has to be done.
+  std::vector<std::vector<std::string>> input(dimension.size());
+  for (size_t i = 0; i < dataset.size(); i++)
+  {
+    col = 0;
+    for (size_t j = 0; j < dataset[i].size(); j++)
+    {
+      if (col < dimension.size())
+      {
+        if (dimension[col] == j)
+        {
+          // insert into input
+          input[col].push_back(dataset[i][j]);
+          col++;
+        }
+      }
+      else
+      {
+        break;
+      }
+    }
+  }
+  data::StringCleaning obj;
+  if (CLI::HasParam("lowercase"))
+  {
+    for (size_t i = 0; i < input.size(); i++)
+      obj.LowerCase(input[i]);
+  }
+  if (CLI::HasParam("stopwords"))
+  {
+    // Not sure how to take input for tokenizer from cli.
+    if (!CLI::HasParam("stopwordsfile"))
+    {
+      throw std::runtime_error("Please provide a file for stopwords.");
+    }
+    ifstream stopwordfile;
+    // Open an existing file
+    const std::string stopwordfilename = CLI::GetParam<std::string>
+        ("stopwordsfile");
+    stopwordfile.open(stopwordfilename);
+    if (!stopwordfile.is_open())
+    {
+      throw std::runtime_error("Unable to open the file for stopwords.");
+    }
+    std::string word;
+    std::deque<std::string> originalword;
+    std::unordered_set<boost::string_view,
+        boost::hash<boost::string_view> >stopwords;
+    while (stopwordfile >> word)
+    {
+      originalword.push_back(word);
+      stopwords.insert(originalword.back());
+    }
+    const std::string delimiter = CLI::GetParam<std::string>
+        ("delimiter");
+    for (size_t i   = 0; i < input.size(); i++)
+      obj.RemoveStopWords(input[i], stopwords,
+          data::SplitByChar(delimiter));
+  }
+  if (CLI::HasParam("punctuation"))
+  {
+    for (size_t i = 0; i < input.size(); i++)
+      obj.RemovePunctuation(input[i]);
+  }
+  const std::string filename2 = CLI::GetParam<std::string>("preprocess"
+      "_dataset");
+  ofstream fout;
+  fout.open(filename2, ios::trunc);
+  if (!fout.is_open())
+  {
+    throw std::runtime_error("Unable to open a file for writing output.");
+  }
+  col = 0;
+  for (size_t i = 0 ; i < dataset.size(); i++)
+  {
+    col = 0;
+    for (size_t j =0 ; j < dataset[i].size(); j++)
+    {
+      if (col < dimension.size() && dimension[col] == j)
+      {
+        fout<<input[col][i]<<column_delimiter;
+        col++;
+      }
+      else
+      {
+        fout<<dataset[i][j]<<column_delimiter;
```

Comment:
Looks like the last delimiter is redundant (you should check that `j+1 < dataset[i].size()`).